client/app/vps/cloud-database/order/vps-cloud-database-order.controller.js

```diff
@@ -0,0 +1,190 @@
+      .then((capacity) => {
+        this.versions = _.map(_.sortBy(capacity.version), version => ({
+          value: version,
+          name: this.$translate.instant(`common_database_version_${version.replace('.', '')}`),
+        }));
+        this.ramAmounts = _.map(capacity.ram, ram => ({
+          value: ram,
+          name: `${ram} ${this.$translate.instant('unit_size_MB')}`,
+        }));
+        this.datacenters = _.map(capacity.datacenter, datacenter => ({
+          value: datacenter,
+          name: this.$translate.instant(`common_datacenter_${datacenter}`),
+        }));
+      })
+      .catch(error => this.CloudMessage.error([
+        this.$translate.instant('vps_tab_cloud_database_order_fetch_capacities_failed'),
+        _(error).get('data.message', error),
+      ].join(' ')));
+  }
+
+  showOrRefreshDurations() {
+    this.cancelFurtherChoices();
+
+    if (!this.currentOrder.version
+            || !this.currentOrder.ram
+            || !this.currentOrder.datacenter) {
+      return this.$q.when();
+    }
+
+    const version = this.currentOrder.version.value;
+    const ram = this.currentOrder.ram.value;
+    const datacenter = this.currentOrder.datacenter.value;
+
+    this.loading.durations = true;
+    return this.ApiOrderDb.getNew({ version, ram, datacenter }).$promise
+      .then((durations) => {
+        this.durations = _.map(durations, duration => ({ value: duration, prices: null }));
+        // we run this in parallel, so no return
+        this.getPricesForEachDuration(this.durations, version, ram, datacenter);
+      })
+      .catch(error => this.CloudMessage.error([
+        this.$translate.instant('vps_tab_cloud_database_order_fetch_duration_failed'),
+        _(error).get('data.message', error),
+      ].join(' ')))
+      .finally(() => {
+        this.loading.durations = false;
+      });
+  }
+
+  cancelFurtherChoices() {
+    this.$timeout.cancel(this.redirectionTimeout);
+    this.loading.purchaseOrder = false;
+    this.loading.redirection = false;
+
+    this.currentOrder.duration = null;
+    this.currentOrder.contractsAccepted = false;
+    this.purchaseOrder = null;
+  }
+
+  getPricesForEachDuration(durations, version, ram, datacenter) {
+    return this.$q.all(_.map(
+      durations,
+      duration => this.getPrices(duration, version, ram, datacenter),
+    )).catch(error => this.CloudMessage.error([
+      this.$translate.instant('vps_tab_cloud_database_order_fetch_prices_failed'),
+      _(error).get('data.message', error),
+    ].join(' ')));
+  }
+
+  getPrices(duration, version, ram, datacenter) {
+    return this.ApiOrderDb.getNewDetails({
+      duration: duration.value, version, ram, datacenter,
+    }).$promise
+      .then((details) => {
+        // we actually want to trigger a change in the UI
+        // by assigning prices to each already shown duration
+        duration.details = details; // eslint-disable-line no-param-reassign
```

Comment:
I think you can use `_.set` to avoid disabling `eslint`

Reply:
We should avoid disabling `eslint`. @marie-j's solution is the right one.

Reply:
No, both solutions are wrong. Either remove the rules or delete them entirely is they shouldn't be followed.

Return the new duration:

```js
.then((details) => Object.assign({}, duration, { details })) // Object.assign mutates the first parameter
```